infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationService.kt

```diff
@@ -785,58 +784,131 @@ class PublicationService @Autowired constructor(
         return asCsvFile(orderedPublishedItems, timeZone ?: ZoneId.of("UTC"))
     }
 
-    fun diffTrackNumber(newTrackNumber: TrackLayoutTrackNumber, oldTrackNumber: TrackLayoutTrackNumber?) =
-        listOf(
+    fun diffTrackNumber(newTrackNumber: TrackLayoutTrackNumber, oldTrackNumber: TrackLayoutTrackNumber?): List<PublicationChange> {
+        val newReferenceLine = newTrackNumber.version?.let { version ->
+            referenceLineService.getByTrackNumberAtMoment(
+                OFFICIAL,
+                newTrackNumber.id as IntId<TrackLayoutTrackNumber>,
```

Comment:
Näissä casteissa ei taida tarvita tuota tyypitystä mainita, koska jo kantaluokka määrää sen. Eli pitäisi riittää `as IntId`

Reply:
Also, my bad että näitä tarvitsee edes tehdä. Tuo DomainId oli vähän osittainen ratkaisu ikuisuusongelmaan asiasta joka on vs ei ole kannassa ja se tapaa pullahtaa esiin. Pitäisi ehkä tehdä ihan omat tyypit niille olioille jotka haetaan kannasta ja niille jotka kirjoitetaan sinne. Kompositiolla se ei välttämättä olisi edes paha homma, ja silloin kannasta haetun tavaran kanssa ei tarvitse arpoa onko sillä kanta-id vai ei.

Reply:
Tein tämän tutkimisesta tiksun, koska asia on ärsyttänyt ennenkin: https://extranet.vayla.fi/jira/browse/GVT-2013